app/services/scholars_archive/caching_service.rb

```diff
@@ -0,0 +1,21 @@
+require 'net/http'
+
+module ScholarsArchive
+  class CachingService
+    def self.fetch_or_store_in_cache(vocabulary)
+      Rails.cache.fetch("/caching/#{vocabulary}", expires_in: 1.day, race_condition_ttl: 15.seconds) do
+        get_jsonld(vocabulary)
```

Comment:
I like the 1.day expiration, seems like the expiration should be an argument to this method instead.. possibly the race condition too.

Since CachingService could be generic and fetch URIs, you could change `vocabulary` to `uri`, and `#get_jsonld` to `#get_response` .. that way the class is general and clear with its responsibilities.